geometry:
  cylinder_height: 0.1
  cylinder_radius: 0.05
  cylinder_mass: 1    # ideally equal to the object
  cylinder_z: 0.25 # table_z + 1/2(table_height+cylinder_height)
  table_position: [0.85,0.0,0.0]
  target_pos: [0.8, 0.0, 0.1]
  friction: 0.005

setup_config:
  fixed_base: true
  update_geometry: true

dynamics:
  dt: 0.015
  initial_state: [ -1.13, 0.39, -0.05, -2.61, 1.26, 1.94, -2.34, 0.04, 0.04, # robot position 9
                   0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,              # robot velocity 9
                   0.42, -0.24, 0.1, 0.0, 0.0, 0.0, 1,                       # mug 0 pose 7
                   0.0, 0.0, 0.0, 0.0, 0.025, 0.5, 0.0,                      # mug 0 velocity 7
                   0]                                                    # contact flag
  gains:
    base_gains:
      kp: []
      kd: []
      ki: []
      i_max: []
    arm_gains:
      kp: [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ]
      kd: [ 30.0, 30.0, 30.0, 30.0, 10.0, 10.0, 1.0 ]
      ki: [ 200.0, 200.0, 200.0, 200.0, 100.0, 100.0, 10.0 ]
      i_max: [ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 ]
    gripper_gains:
      kp: [ 100, 100]
      kd: [ 50, 50 ]
      ki: [ 0.0, 0.0 ]
      i_max: [ 0.0, 0.0 ]

sim_dynamics:
  dt: 0.015
  initial_state: [ -1.13, 0.39, -0.05, -2.61, 1.26, 1.94, -2.34, 0.04, 0.04, # robot position 9
                   0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,              # robot velocity 9
                   0.42, -0.24, 0.1, 0.0, 0.0, 0.0, 1,                       # obj_0 pose 7
                   0.0, 0.0, 0.0, 0.0, 0.025, 0.5, 0.0,                      # obj_0 velocity 7
                   0]                                                        # contact flag

  gains:
    base_gains:
      kp: []
      kd: []
      ki: []
      i_max: []
    arm_gains:
      kp: [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 ]
      kd: [ 30.0, 30.0, 30.0, 30.0, 10.0, 10.0, 1.0 ]
      ki: [ 200.0, 200.0, 200.0, 200.0, 100.0, 100.0, 10.0 ]
      i_max: [ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 ]
    gripper_gains:
      kp: [ 100, 100 ]
      kd: [ 50, 50 ]
      ki: [ 0.0, 0.0 ]
      i_max: [ 0.0, 0.0 ]

cost:
  regularization: 0.0
  linear_weight: 250.0
  linear_weight_opening: 50.0

  angular_weight: 0.0 #45.0
  angular_weight_opening: 1.0

  min_dist: 0.0
  max_reach: 3.0
  reach_weight: 0
  reach_weight_slope: 0

  obstacle_weight: 0
  obstacle_weight_slope: 0
  obstacle_radius: 0.8

  contact_weight: 10000.0

  object_weight: 80
  object_tolerance: 0.01

  grasp_translation_offset: [ 0.0, 0.0, 0.0 ]
  grasp_orientation_offset: [ 0, -0.7071068, 0, 0.7071068 ]    # -90deg around y

  upper_joint_limits: [ 6.28, 2.8973, 1.7628, 2.8973, 0.0698, 2.8973, 3.7525, 2.8973 ]
  lower_joint_limits: [ -6.28, -2.8973, -1.7628, -2.8973, -3.0718, -2.8973, -0.0175, -2.8973 ]

  joint_limit_cost: 0 #1000
  joint_limit_slope: 0 #100

  max_power: 0.0
  power_weight: 10.0

  tracked_frame: panda_grasp
  arm_base_frame: panda_link0

  collision_link_0: panda_link1
  collision_link_1: panda_link7
  collision_threshold: 0.35
  collision_weight: 0
